kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -64,11 +64,41 @@ public SnapshotImpl(
     this.inCommitTimestampOpt = Optional.empty();
   }
 
+  /////////////////
+  // Public APIs //
+  /////////////////
+
   @Override
   public long getVersion(Engine engine) {
     return version;
   }
 
+  /**
+   * Get the timestamp (in milliseconds since the Unix epoch) of the latest commit in this Snapshot.
+   *
+   * <p>When InCommitTimestampTableFeature is enabled, the timestamp is retrieved from the
+   * CommitInfo of the latest commit in this Snapshot, which can result in an IO operation.
+   *
+   * <p>For non-ICT tables, this is the same as the file modification time of the latest commit in
+   * this Snapshot.
+   */
```

Comment:
If we don't think any external developers will ever see an uninitialized snapshot (empty table) can we at least clarify the behavior here for that case?